Trezor encrypt

petrkr opened the pull request:
###### Motivation for this change
Make new version of old PR trezorCryptKeyValue. This version uses latest official Trezord-go code base and it does not use 3rd party libraries for its work. As main it works with both Trezor one and Trezor T

###### Things done

Prepare LUKS environment
Maintainer
Added initial version of new trezorencrypt as replacement of trezorCryptKeyValue

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

petrkr committed:
maintainers: Add petrkr (Petr Kracik)

petrkr commented:
Rebase to latest master. Updated version 0.1.1 with some fixes about exit codes

jb55 commented on pkgs/tools/misc/trezorencrypt/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+buildGoPackage rec {
+  name = "trezorencrypt-${version}";
+  version = "v0.1";
```
should be `version = "0.1"`

jb55 commented on pkgs/tools/misc/trezorencrypt/default.nix:
```diff
@@ -0,0 +1,33 @@
+  src = fetchFromGitHub {
+    owner  = "petrkr";
+    repo   = "trezorencrypt";
+    rev    = "${version}";
```
`rev = "v${version}"`

petrkr replied:
So name of nix's package will not contain that 'v' ?right?

jb55 replied:
correct, nixpkgs has version functions that expect the `0.1` form as well.

jb55 reviewed (changes requested):
This looks neat, looking forward to trying it out. Just a few nits.

jb55 commented:
was 14443df793ee806187f51e87c2223ab6a706f5d9 meant to be included in this?

jb55 commented:
also please rebase+squash the last two commits into `trezorencrypt: init at 0.1.1`

petrkr commented:
First i think to separate to another PR, i just before had whole PR and this is not breaking but it does not corespond with trezor encrypt package, i'll do separate then later

Squashing I can do, since initial version was not merged so have extra commit about change version is useless, i agree.

Regards

petrkr committed:
trezorencrypt: Initial package commit at 0.1.2

petrkr commented:
Rebased to latest master, included notes from discuss.

jb55 reviewed (approved):
LGTM

jb55 commented:
I tested this with my trezor and it works great. I would merge but I don't have the ability to do so :(

petrkr commented:
Found one bug in versiosn 0.1.1, so released 0.1.2. There is bug if you do not have set trezor's label. I will crash...

So now I rebased to latest master and squashed initial release to 0.1.2. You can test old and new as well.
